player/public/coreplugins/org.ekstep.videorenderer-1.1/renderer/plugin.js

```diff
@@ -0,0 +1,324 @@
+/**
+ * @description Launcher to render the Video or youtube URL's
+ * @extends {class} org.ekstep.contentrenderer.baseLauncher
+ * @author Gourav More <gouav_m@tekditechnologies.com>
+ */
+
+org.ekstep.contentrenderer.baseLauncher.extend({
+    _time: undefined,
+    supportedStreamingMimeType: "application/x-mpegURL",
+    isStreaming: EkstepRendererAPI.getGlobalConfig().streaming || false,
+    messages: {
+        noInternetConnection: "Internet not available. Please connect and try again.",
+        unsupportedVideo: "Video URL not accessible"
+    },
+    currentTime: 1,
+    videoPlayer: undefined,
+    stageId: undefined,
+    heartBeatData:{},
+    enableHeartBeatEvent:false,
+    _constants: {
+        mimeType: ["video/mp4", "video/x-youtube", "video/webm"],	
+        events: {	
+            launchEvent: "renderer:launch:video"	
+        }	
+    },	
+    initLauncher: function() {	
+        EkstepRendererAPI.addEventListener(this._constants.events.launchEvent, this.start, this);
+        EkstepRendererAPI.addEventListener("renderer:overlay:mute", this.onOverlayAudioMute, this);
+        EkstepRendererAPI.addEventListener("renderer:overlay:unmute", this.onOverlayAudioUnmute, this);
+    },
+    start: function() {
+        this._super();
+        var data = _.clone(content);
+        this.heartBeatData.stageId = content.mimeType === 'video/x-youtube' ? 'youtubestage' : 'videostage';
+        var globalConfigObj = EkstepRendererAPI.getGlobalConfig();
+        if (window.cordova || !isbrowserpreview) {
+            var prefix_url = globalConfigObj.basepath || '';
+            path = prefix_url + "/" + data.artifactUrl;
+        } else {
+            path = data.artifactUrl;
+        }
+        this.createVideo(path, data);
+        this.configOverlay();
+    },
+    createVideo: function(path, data) {
+        EkstepRendererAPI.dispatchEvent("renderer:splash:hide");
+
+        video = document.createElement('video-js');
+        video.style.width = '100%';
+        video.style.height = '100%';
+        video.style.position = 'absolute';
+        video.id = "videoElement";
```

Comment:
don't use generic-name as a identifier `videoElement ` better you should user `this.manifest.id` as a video element identifier